<?xml version="1.0"?>
<launch>
  <arg name="config_path"/>
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb --args" />
  <arg name="save_data" default="false" />
  <arg name="save_location" default="$(env HOME)/.ros/" />

	<!--core coordinating service-->
  <node name="framefab_core_service" pkg="framefab_core" type="framefab_core_service" output="screen" required="true" launch-prefix="$(arg launch_prefix)">
    <rosparam command="load" file="$(arg config_path)/model_input.yaml"/>
    <rosparam command="load" file="$(arg config_path)/path_input.yaml"/>
    <rosparam command="load" file="$(arg config_path)/robot_input.yaml"/>
    <rosparam command="load" file="$(arg config_path)/output_path_input.yaml"/>
    <param name="save_data" value="$(arg save_data)" />
    <param name="save_location" value="$(arg save_location)"/>
  </node>

	<!--task sequence formatting service-->
  <node name="framefab_task_sequence_processor_node" pkg="framefab_task_sequence_processor" type="task_sequence_processor_node" output="screen" required="true"/>

	<!--output trajectory formatting service-->
  <node name="framefab_output_processor_node" pkg="framefab_output_processor" type="framefab_output_processor_node" output="screen" required="true"/>

</launch>
